src/components/FileRepo/FileRepo.js

```diff
@@ -220,13 +220,13 @@ const AggregationsWrapper = compose(injectState, withTheme)(
                           }
                         },
                         onSqonSubmit: ({ sqon }) => {
-                          setSQON(sqon);
+                          effects.unsetModal();
+                          setTimeout(() => setSQON(sqon), 0);
```

Comment:
let's move this inside the unsetModal as a callback so we can minimize the use of this pattern :P (Jeff is watching!! :)))